3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,60 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
-}
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+async function fetchData(url) {
   // TODO complete this function
+  const response = await fetch(url);
```

Comment:
You should also check for HTTP error, e.g.:

```js
if (!response.ok) {
   throw new Error(...)
}
```

Reply:
@are you sure about this one Jim?
i tried it
even if response is 404 it doesn't throw a new error
but goes to the .catch and console.logs whatever is in the .catch

Reply:
So where is the changed code that you tried? I can find it in this PR.